### PR TITLE
Testing: Added ARP entries to Mozilla Firefox 98.0.2.

### PR DESCRIPTION
**Pull request by jedieaston**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

I made a [thing](https://github.com/jedieaston/Add-ARPEntries) to generate ARP entries by running the installers in Docker containers. This is a test to make sure the manifest validates okay.

btw: since there's no containers on ARM64, I couldn't fill those ones in. For some apps it can be puzzled together, but for most it's hard to get accurate info without running the installer.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/55957)

### CODE REVIEW

**Review by OfficialEsco — APPROVED** [review]
Hopefully this resolves the weird issue where it doesn't match unless you update it via WinGet..

**Comment by OfficialEsco** [comment]
Do you know if the ARP matching is in the 1.3.692-preview version?
I'm sadly still seeing
```
PS M:\> winget list
Name                                        Id                                                 Version          Available
----------------------------------------------------------------------------------------------------------------------------
Mozilla Firefox (x64 en-US)                 Mozilla Firefox 98.0.2 (x64 en-US)                 98.0.2
```
when its manually installed, maybe we have to remove/fix the ProductCode since the Package don't actually use the MSI ProductCode?

**Comment by jedieaston** [comment]
Still matching the same way. The ProductCode field inside AppsAndFeaturesEntries isn't being used for matching yet though, so maybe we should wait for that before removing the other ProductCode?

File: manifests/m/Mozilla/Firefox/98.0.2/Mozilla.Firefox.installer.yaml [hunk 1]
```diff
@@ -27,6 +25,15 @@ Installers:
   InstallerUrl: https://download-installer.cdn.mozilla.net/pub/firefox/releases/98.0.2/win64/en-US/Firefox%20Setup%2098.0.2.msi
   InstallerSha256: AC31A566E82736073AEBF230B9557AF1AB43514C7A5C2149DC78FF98912AA9C9
   ProductCode: '{1294A4C5-9977-480F-9497-C0EA1E630130}'
```
Comment (OfficialEsco) [inline manifests/m/Mozilla/Firefox/98.0.2/Mozilla.Firefox.installer.yaml:27]:
```suggestion
  ProductCode: 'Mozilla Firefox 98.0.2 (x64 en-US)'
```
I mean change it to like this which is the real ProductCode for en-US Firefox x64

Comment (jedieaston) [reply manifests/m/Mozilla/Firefox/98.0.2/Mozilla.Firefox.installer.yaml:27]: Oh, that makes more sense.

**Pull request merged**